src/Adaptivity/AdaptedDiscreteModels.jl

```diff
@@ -0,0 +1,281 @@
+
+"""
+  AdaptedDiscreteModel
+
+  `DiscreteModel` created by refining/coarsening another `DiscreteModel`. 
+  The refinement/coarsening hierarchy can be traced backwards by following the 
+  `parent` pointer chain. This allows the transfer of dofs 
+  between `FESpaces` defined on this model and its ancestors.
+
+"""
+struct AdaptedDiscreteModel{Dc,Dp,A<:DiscreteModel{Dc,Dp},B<:DiscreteModel{Dc,Dp},C<:AdaptivityGlue} <: DiscreteModel{Dc,Dp}
+  model  ::A
+  parent ::B
+  glue   ::C
+
+  function AdaptedDiscreteModel(model::DiscreteModel{Dc,Dp},parent,glue) where {Dc,Dp}
+    @check !isa(model,AdaptedDiscreteModel)
+    A = typeof(model)
+    B = typeof(parent)
+    C = typeof(glue)
+    return new{Dc,Dp,A,B,C}(model,parent,glue)
+  end
+end
+
+# DiscreteModel API
+Geometry.get_grid(model::AdaptedDiscreteModel)          = get_grid(model.model)
+Geometry.get_grid_topology(model::AdaptedDiscreteModel) = get_grid_topology(model.model)
+Geometry.get_face_labeling(model::AdaptedDiscreteModel) = get_face_labeling(model.model)
+
+# Other getters
+get_model(model::AdaptedDiscreteModel)  = model.model
+get_parent(model::AdaptedDiscreteModel{Dc,Dp,A,<:AdaptedDiscreteModel}) where {Dc,Dp,A} = get_model(model.parent)
+get_parent(model::AdaptedDiscreteModel{Dc,Dp,A,B}) where {Dc,Dp,A,B} = model.parent
+get_adaptivity_glue(model::AdaptedDiscreteModel) = model.glue
+
+# Relationships
```

Comment:
Add a docstring here. 
```julia
"""
Returns true if m1 is a "child" model of m2, i.e., if m1 is the result of adapting m2
""""
```

Reply:
We have the `is_child` method. Ok. 

Why don't we also add the `is_parent` method?

Reply:
Done!